exonum/src/node/basic.rs

```diff
@@ -49,20 +45,20 @@ where
         }
     }
 
-    /// Handles the `Connected` event. Node's `Connect` message is sent as response.
-    pub fn handle_connected(&mut self, addr: &SocketAddr) {
-        info!("Connected to: {}", addr);
-        let message = self.state.our_connect_message().clone();
-        self.send_to_addr(addr, message.raw());
+    /// Handles the `Connected` event. Node's `Connect` message is sent as response
+    /// if received `Connect` message is correct.
+    pub fn handle_connected(&mut self, addr: SocketAddr, connect: Connect) {
+        info!("Received Connect message from peer: {}", addr);
```

Comment:
@vldm What do you think about passing `SocketAddr` by value here (and below)? :smile:

Reply:
I am sure `SocketAddr` is a small enum to pass it by value.

Reply:
Similar to `PublicKey`? :smile:

Reply:
https://github.com/exonum/exonum/issues/167
Hm...